HandheldCompanion/Managers/InputsManager.cs

```diff
@@ -158,6 +158,24 @@ private static bool CheckForSequence(bool IsKeyDown, bool IsKeyUp)
                     if (hotkey is null)
                         continue;
 
+                    // special case HIDmode switch hotkey
+                    // required here in order to (immediatly) disable hotkey while HIDmode is being changed to avoid duplicates
+                    // this takes care of repeated keybinds actions
+                    if (hotkey.Listener == "shortcutChangeHIDMode")
+                    {
+                        var inputType = currentChord.InputsType;
+                        if ((inputType == InputsChordType.Click && IsKeyUp) || (inputType == InputsChordType.Long && IsKeyDown))
+                        {
+                            var hidHotkeys = HotkeysManager.Hotkeys.Values.Where(item => item.inputsHotkey.Listener.Equals("shortcutChangeHIDMode"));
+                            foreach (var hidHotkey in hidHotkeys)
+                            {
+                                if (!hidHotkey.IsEnabled)
+                                    return false;
+                                System.Windows.Application.Current.Dispatcher.Invoke(() => { hidHotkey.IsEnabled = false; });
+                            }
+                        }
+                    }
```

Comment:
Wouldn't it be better to use BeginInvoke? 
Invoke makes the current calling thread wait for this to happen. 

https://stackoverflow.com/questions/229554/whats-the-difference-between-invoke-and-begininvoke

Reply:
Intended behavior is to block subsequent code from executing until the UI change has taken effect to avoid button click while keybind hotkey is being run, causing a duplicate call; hence Invoke might be the better option here.